### PR TITLE
SDIT-2319 Separate filters into another file and add filter tests

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: integration_tests/pages/search.ts [hunk 1]
```diff
@@ -33,6 +33,16 @@ export default class Search extends Page {
     return this.searchButton().click()
   }
 
+  maleFilter = (): PageElement => cy.contains('.moj-filter__tag', 'Male')
+
+  femaleFilter = (): PageElement => cy.contains('.moj-filter__tag', 'Female')
+
+  hdcFilter = (): PageElement => cy.contains('.moj-filter__tag', 'HDC')
```
Comment (karenmillermoj) [inline integration_tests/pages/search.ts:40]: I can modify this to pass thru text for filter - will change in next PR

File: server/routes/search/searchController.ts [hunk 2]
```diff
@@ -62,7 +62,6 @@ export default class SearchController {
     req.session.searchParams ??= {}
 
     req.session.prisonerSearchForm = { ...trimForm(req.body) }
-    logger.debug('search form is', req.session.prisonerSearchForm)
```
Comment (karenmillermoj) [inline server/routes/search/searchController.ts:65]: This controller still has no separate tests - I will add these in

File: server/routes/search/searchController.ts [hunk 3]
```diff
@@ -124,7 +126,7 @@ export default class SearchController {
     req.session.searchParams.filters?.forEach(entry => {
       searchParams.append('filters', entry)
     })
-    return searchParams ? '' : searchParams.toString()
+    return searchParams ? searchParams.toString() : ''
   }
```
Comment (andymarke) [inline server/routes/search/searchController.ts:130]: I guess you could just do

`return searchParams?toString() ?? '' `

**Review by andymarke — APPROVED** [review]
lgtm

File: server/routes/search/searchController.ts (hunk 3 above)
Comment (karenmillermoj) [reply server/routes/search/searchController.ts:130]: will add in next PR

**Pull request merged**